src/components/Leaderboard.vue

```diff
@@ -25,54 +29,49 @@ const {
 	return resp.data
 }, null)
 
-const {
-	reveal: revealDelete,
-	isRevealed: isRevealedDelete,
-	cancel: cancelDelete
-} = useConfirmDialog()
-
-const {
-	reveal: revealRestore,
-	isRevealed: isRevealedRestore,
-	cancel: cancelRestore
-} = useConfirmDialog()
-
-async function confirmDeleteBoard() {
-	// TODO: Error-handling
-	await leaderboards.deleteLeaderboard(props.id, useAuth(token.value))
-	execute()
+async function revealDelete() {
+	if (
+		confirm('Really delete this leaderboard? (This action can be reversed)')
+	) {
+		useApi(
+			() => leaderboards.deleteLeaderboard(props.id, useAuth(token.value)),
+			() => execute(),
+			(error) => {
+				updateError.value =
+					'Failed to delete: ' + (error as Response).status.toString(10)
+			}
+		)
+	}
 }
 
-async function confirmRestoreBoard() {
-	// TODO: Error-handling
-	await leaderboards.restoreLeaderboard(props.id, useAuth(token.value))
-	execute()
+async function revealRestore() {
+	if (
+		confirm('Really restore this leaderboard? (This action can be reversed)')
+	) {
+		useApi(
+			() => leaderboards.restoreLeaderboard(props.id, useAuth(token.value)),
+			() => execute(),
+			(error) => {
+				updateError.value =
+					'Failed to restore: ' + (error as Response).status.toString(10)
+			}
+		)
+	}
 }
 </script>
 
 <template>
 	<div class="container">
 		<div v-if="isLoading">Loading...</div>
 		<div v-else-if="error" class="error-container">
-			<p>Error</p>
+			<p class="errorText">{{ error }}</p>
```

Comment:
The error object isn't going to print nicely. It'll just give you some [object Object] type nonsense.